pkg/test/e2e/ccm-migration/ccm_migration_test.go

```diff
@@ -40,117 +40,156 @@ import (
 )
 
-const (
-	userClusterPollInterval = 5 * time.Second
-	customTestTimeout       = 10 * time.Minute
-)
-
 var _ = ginkgo.Describe("CCM migration", func() {
-	ctx := context.Background()
-
 	var (
-		userClient ctrlruntimeclient.Client
-		clusterJig *ClusterJig
+		seedClient            ctrlruntimeclient.Client
+		userClient            ctrlruntimeclient.Client
+		clusterClientProvider *clusterclient.Provider
+		clusterJig            providers.ClusterJigInterface
+
+		err error
 	)
 
-	ginkgo.Context("supported provider", func() {
-		ginkgo.BeforeEach(func() {
-			seedClient, _, _ := e2eutils.GetClientsOrDie()
-			clusterClientProvider, err := clusterclient.NewExternal(seedClient)
-			gomega.Expect(err).NotTo(gomega.HaveOccurred())
-
-			clusterJig = &ClusterJig{
-				Log:            e2eutils.DefaultLogger,
-				SeedClient:     seedClient,
-				Name:           options.userClusterName,
-				DatacenterName: options.datacenter,
-				Version:        options.kubernetesVersion,
-			}
+	ginkgo.BeforeEach(func() {
+		seedClient, _, _ = e2eutils.GetClientsOrDie()
+		clusterClientProvider, err = clusterclient.NewExternal(seedClient)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	})
 
-			gomega.Expect(clusterJig.SetUp(ctx, kubermaticv1.CloudSpec{
-				Openstack: &kubermaticv1.OpenstackCloudSpec{
-					FloatingIPPool: options.osCredentials.floatingIPPool,
-					CredentialsReference: &machinecontrollertypes.GlobalSecretKeySelector{
-						ObjectReference: corev1.ObjectReference{
-							Name:      fmt.Sprintf("credential-openstack-%s", clusterJig.Name),
-							Namespace: resources.KubermaticNamespace,
-						},
-					},
-				},
-				DatacenterName: options.datacenter,
-			}, options.osCredentials)).NotTo(gomega.HaveOccurred(), "user cluster should deploy successfully")
-			clusterJig.Log.Debugw("Cluster set up", "name", clusterJig.Cluster.Name)
-
-			gomega.Expect(wait.Poll(userClusterPollInterval, customTestTimeout, func() (done bool, err error) {
-				userClient, err = clusterClientProvider.GetClient(context.Background(), clusterJig.Cluster)
-				if err != nil {
-					clusterJig.Log.Debug("user cluster client get failed, retrying...")
-					return false, nil
-				}
-				return true, nil
-			})).NotTo(gomega.HaveOccurred())
-			gomega.Expect(clusterv1alpha1.AddToScheme(userClient.Scheme())).NotTo(gomega.HaveOccurred())
+	ginkgo.Context("vSphere provider", func() {
+		var (
+			vsphereCluster *kubermaticv1.Cluster
+		)
 
-			gomega.Expect(clusterJig.CreateMachineDeployment(ctx, userClient, options.osCredentials))
-			clusterJig.Log.Debug("MachineDeployment created")
+		ginkgo.BeforeEach(func() {
+			if options.provider != string(kubermaticv1.VSphereCloudProvider) {
+				ginkgo.Skip("skipping openstack test")
+			}
+			vsphereCluster = &kubermaticv1.Cluster{}
+			clusterJig = providers.NewClusterJigVsphere(seedClient, options.kubernetesVersion, options.vsphereSeedDatacenter, options.vSphereCredentials)
```

Comment:
This all feels a bit repetitive. Can we just provide the cluster type (vSphere or OpenStack) to this function and expect it to return a proper cluster?